pywinauto/controls/uia_controls.py

```diff
@@ -938,9 +938,27 @@ def __init__(self, elem):
         super(MenuItemWrapper, self).__init__(elem)
 
     # -----------------------------------------------------------
+    _cache_items = {}
     def items(self):
         """Find all items of the menu item"""
-        return self.children(control_type="MenuItem")
+        items = self.children(control_type="MenuItem")
+
+        # If a menu is not displayed then the UIA controls (children of self)
+        # will not yet exist. We can bring them into existence by clicking the
+        # menu (self). The controls can also (mysteriously) lose all their
+        # element_info. So once we see them bv clicking the menu we cache them
+        # for subsequent use.
+        if not items:
+            if self._cache_items.get(self.element_info.automation_id, None):
```

Comment:
I wouldn't rely on `automation_id` property as a key of dictionary because it's possible to have child items without such property. I can't provide an example right now because of being too busy, but it's worth trying different apps with this approach to test everything works the same way or better.

Reply:
Wow, time flies. I too am only looking at this again now. If `automation_id` is not reliable we could just check that it is available and if it is, then cache the item.

Reply:
Never say "Never!". :) Yeah, that may be better, but why not use `self.element_info.name`? Maybe it's a specific menu implementation. Is it Qt5 app?

Reply:
Nope a VCL (Borland) app. I have of course since discovered how neat pywinauto's implementation of metaclasses is such that I simply define a new class derived from this one, and override items, and that is respected and used by pywinauto (the metaclass updates a dict mapping of controls to wrappers). Which means I'm actually free to experiment willy-nilly on broader impacts. Part of me thinks the earlier solution I suggest of an optional argument click_to_find_children=False is impact neutral on existing functionality and does nothing special.

Reply:
Thanks for the details! I will think what we can do.

Reply:
Turns out rather that `automation_id` which indeed proved unreliably available, the only key I could find for a caching dict is `element_info.element` which seems to be a unique pointer (which Python permits as a dict key). Problem is nothing else I could find anywhere under `element_info` provides a unique handle on the element (include `name` which is blank on most of our controls).